### PR TITLE
ROSAENG-64850 | rosa-day1-negative-f7 | fix(e2e): align registry asserts and retry AWS region flakes

### PR DESCRIPTION
**Pull request by amandahla**
## PR Summary

Align registry-config error assertions with lowercased CLI output (#3419), migrate OCP-76396 checks to `helper.ExpectErrorWithMessage`, and retry transient AWS regions/credentials failures in `CreateDryRun` so day1-negative specs reach their intended validation errors.

## Detailed Description of the Issue

Two E2E stability issues in `rosa-day1-negative-f7`:

1. **Registry config error casing** — PR #3419 lowercased registry-config validation messages in the CLI, but OCP-76396 / OCP-77149 assertions still expected title-case strings.
2. **AWS region/credential flakes** — `CreateDryRun` calls `GetRegionList` early; transient `Failed to retrieve AWS regions` / credential validation errors can fail negative specs before they reach the validation error under test (e.g. OCP-66761).

## Related Issues and PRs

- Jira: [ROSAENG-64850](https://issues.redhat.com/browse/ROSAENG-64850), [OCP-76396](https://issues.redhat.com/browse/OCP-76396)
- Related PR(s): #3419 (lowercased registry-config errors)

## Type of Change

- [x] fix - resolves incorrect behavior or test flakes.
- [ ] feat - adds a new user-facing capability.
- [ ] docs - updates documentation only.
- [ ] style - formatting or naming changes with no logic impact.
- [ ] refactor - code restructuring with no behavior change.
- [ ] test - adds or updates tests only.
- [ ] chore - maintenance work (tooling, housekeeping, non-product code).
- [ ] build - changes build system, packaging, or dependencies for build output.
- [ ] ci - changes CI pipelines, jobs, or automation workflows.
- [ ] perf - improves performance without changing intended behavior.

## Previous Behavior

- OCP-76396 / OCP-77149 failed on registry-config assertion casing mismatches.
- OCP-76396 used raw `ContainSubstring` on command output for error checks.
- `CreateDryRun` returned immediately on transient AWS regions/credentials failures.

## Behavior After This Change

- Registry error expectations use lowercase strings; OCP-76396 uses `helper.ExpectErrorWithMessage` (case-insensitive) for all checks in that test case.
- `CreateDryRun` retries up to 3 times (10s delay) on transient AWS regions/credentials errors via `RetryOnAWSRegionsCredentialError`.
- Unit tests cover the transient classifier and retry boundary.

### Preconditions

- AWS credentials and OCM token configured
- `TEST_PROFILE` set for e2e (see repo test docs)

### Test Steps

1. `go test ./tests/utils/config/ -count=1`
2. `make fmt-check && make lint && make test`
3. Run OCP-76396 / day1-negative filter locally or via periodic job `rosa-day1-negative-f7`

### Expected Results

- Config unit tests pass (classifier + retry boundary).
- OCP-76396 passes with lowercased registry validation messages.
- Day1-negative specs tolerate transient AWS setup flakes on `CreateDryRun`.

## Proof of the Fix

- `go test ./tests/utils/config/` — classifier and retry unit tests pass.
- `go build ./tests/e2e/...` — compiles.
- Local `day1-negative` run (`day1-negative-ginkgo-2.log`): **29 passed | 3 failed** (~2h24m)
  - **OCP-76396 passed** (registry casing + `ExpectErrorWithMessage`)
  - **OCP-66761 passed** (retry path exercised)
  - Remaining failures: OCP-68971 (`SHARED_VPC_*`), OCP-45509 (proxy/subnet message), OCP-71329 (AWS creds masking label) — out of scope

## Breaking Changes

- [x] No breaking changes
- [ ] Yes, this PR introduces a breaking change (describe impact and migration plan below)

## Developer Verification Checklist

- [x] Commit subject/title follows `[JIRA-TICKET] | [TYPE]: <MESSAGE>`.
- [x] PR description clearly explains both **what** changed and **why**.
- [x] Relevant Jira/GitHub issues and related PRs are linked.
- [x] `make install-hooks` has been run in this clone.
- [x] Tests were added/updated where appropriate.
- [ ] I manually tested the change. (E2E requires OCM/AWS env; local day1-negative run cited above.)
- [x] `make test` passes.
- [x] `make lint` passes.
- [x] `make rosa` passes.
- [ ] Documentation or repo-local agent guidance was added/updated where appropriate.
- [x] Any risk, limitation, or follow-up work is documented.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved cluster creation reliability by automatically retrying transient AWS region and credential errors.
  - Updated registry configuration validation expectations for consistent lowercase error messages.

- **Tests**
  - Added coverage for AWS error detection, retry behavior, successful recovery, exhausted retries, and zero-retry scenarios.
  - Added configuration test-suite registration.

[ROSAENG-64850]: https://redhat.atlassian.net/browse/ROSAENG-64850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/utils/config/aws_retry.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		combined += "\n" + err.Error()
+	}
+	return strings.Contains(combined, "Failed to retrieve AWS regions") ||
+		strings.Contains(combined, "AWS was not able to validate the provided access credentials")
+}
+
+// RetryOnAWSRegionsCredentialError retries fn when the result indicates a
+// transient AWS regions / credentials failure (for example CLUSTERS-MGMT-400
+// while listing regions). Modeled on RetryOnIAMPropagationError.
+func RetryOnAWSRegionsCredentialError(
```
Comment (olucasfreitas) [inline tests/utils/config/aws_retry.go:27]: This helper now sits on the shared `CreateDryRun` path, so it changes the failure-handling contract for a large slice of the negative suite. Can we add focused tests for the transient classifier and retry boundary here before we rely on string-matched retries across all those callers?

**Comment by olucasfreitas** [comment]
I agree with keeping this one open. `RetryOnAWSRegionsCredentialError` is now on the shared `CreateDryRun` path, so a small table-driven test here would protect a lot of negative coverage from silently drifting.

**Comment by olucasfreitas** [comment]
I wouldn't block this PR on a cancelable sleep in the helper. The higher-signal gap is direct coverage of the retry contract itself, since that's the part that now affects a broad slice of suite behavior.

File: tests/utils/config/aws_retry.go (hunk 1 above)
Comment (amandahla) [reply tests/utils/config/aws_retry.go:27]: Good point, I added the tests/utils/config/aws_retry_test.go (with config_suite_test.go)

File: tests/e2e/test_rosacli_cluster.go [hunk 2]
```diff
@@ -3132,7 +3132,7 @@ var _ = Describe("HCP cluster creation negative testing",
 					Expect(output.String()).
 						To(
 							ContainSubstring(
-								"ERR: Setting the registry config is only supported for hosted clusters"))
+								"ERR: setting the registry config is only supported for hosted clusters"))
```
Comment (jerichokeyne) [inline tests/e2e/test_rosacli_cluster.go]: @amandahla Would you mind updating these to use the `helper.ExpectErrorWithMessage` instead of just updating the error messages? I'm trying to slowly migrate all error checks to use that helper function since it should help avoid a lot of the case sensitivity errors we've been running into for a while

Normally when I find a test case that needs an error message updated I try to update *all* the error message checks within that singular test case to use the new helper function

Comment (jerichokeyne) [reply tests/e2e/test_rosacli_cluster.go]: Other than this fairly minor thing, this PR looks good to me

Comment (amandahla) [reply tests/e2e/test_rosacli_cluster.go]: Good suggestion, thanks. Test updated to use the helper!

Comment (olucasfreitas) [reply tests/e2e/test_rosacli_cluster.go]: after the helper update, this look LGTM to me

**Comment by amandahla** [comment]
@coderabbitai why required check test failed?
